include/xml_writer.hpp

```diff
@@ -16,7 +16,7 @@ class xml_writer : public output_writer {
 
   xml_writer(const std::string &, const boost::program_options::variables_map &, const user_map_t &,
              const boost::posix_time::ptime &max_time,
-             bool has_history = false, bool has_changeset_discussions = false);
+             bool clean = false, bool has_history = false, bool has_changeset_discussions = false);
```

Comment:
I'd prefer new default arguments to be appended at the end. This reduces the scope for a change of meaning where invocations specify a few (but not all) arguments. E.g: here, `xml_writer("", vm, user_map, max_time, true)` would mean something different now.

Note that this comment is superseded by one below about moving to `enum class`es anyway.